backend/lib/scheduler/Scheduler.js

```diff
@@ -18,22 +20,16 @@ class Scheduler {
         this.config = options.config;
         this.robot = options.robot;
         this.ntpClient = options.ntpClient;
+        this.checkedTime = null;
```

Comment:
I needed a while to understand what is happening here.
This naming change might help, unless I've misunderstood how this works

```suggestion
        this.nextScheduledCheckTime = null;
```